tools/applicationinsights-web-snippet/src/miniLoader.ts

```diff
@@ -0,0 +1,401 @@
+
+        function _createEnvelope(iKey:string, theType:string) {
+            let tags = {};
+            let type = "Browser";
+            tags[strAiDevice + "id"] = type[strToLowerCase]();
+            tags[strAiDevice + "type"] = type;
+            tags[strAiOperationName] = locn && locn.pathname || "_unknown_";
+            tags[strAiSdkVersion] = "javascript:snippet_" + (appInsights.sv || appInsights.version);
+
+            let envelope:IEnvelope = {
+                time: _getTime(),
+                iKey: iKey,
+                name: "Microsoft.ApplicationInsights." + iKey.replace(/-/g, strEmpty) + "." + theType,
+                sampleRate: 100,
+                tags: tags,
+                data: {
+                    baseData: {
+                        ver: 2
+                    }
+                },
+                ver: 4.0,
+                seq: "1",
+                aiDataContract: undefined
+            };
+
+            return envelope;
+        }
+
+        function _createInternal(iKey:string, message:string, targetSrc:string, endpointUrl:any) {
+            let envelope : IEnvelope = _createEnvelope(iKey, "Message");
+            let data = envelope.data;
+            data.baseType = "MessageData";
+            let baseData = data.baseData;
+
+            baseData.message = "AI (Internal): 99 message:\"" + (message + " (" + targetSrc + ")").replace(/\"/g, strEmpty) + "\"";
+            baseData.properties = {
+                endpoint: endpointUrl
+            };
+            
+            return envelope;
+        }
+
+        function _createException(iKey:string, message:string, targetSrc:string, endpointUrl:any) {
+            let envelope : IEnvelope = _createEnvelope(iKey, "Exception");
+            let data = envelope.data;
+            data.baseType = "ExceptionData";
+            data.baseData.exceptions = [{
+                typeName: "SDKLoadFailed",
+                message: message.replace(/\./g, "-"),  // Replacing '.' characters as it causes the portal to hide the start of the message in the summary
+                hasFullStack: false,
+                stack: message + "\nSnippet failed to load [" + targetSrc + "] -- Telemetry is disabled\nHelp Link: " + helpLink + "\nHost: " + (locn && locn.pathname || "_unknown_") + "\nEndpoint: " + endpointUrl,
+                parsedStack: []
+            }];
+
+            return envelope;
+        }
+
+        let domainRetryIndex = -1;
+        let domainRetryCount = 0;
+        let domains = [
+            "js.monitor.azure.com",
+            "js.cdn.applicationinsights.io",
+            "js.cdn.monitor.azure.com",
+            "js0.cdn.applicationinsights.io",
+            "js0.cdn.monitor.azure.com",
+            "js2.cdn.applicationinsights.io",
+            "js2.cdn.monitor.azure.com",
+            "az416426.vo.msecnd.net" // this domain is supported but not recommended
+        ];
+
+        fetch("https://js.monitor.azure.com/beta/ai.3.integrity.json")
```

Comment:
`fetch` doesn't exist in all runtimes :-(

So we will need to abstract this out to use either
`fetch` or `XMLHttpRequest`, lets not add support for IE (which means we would also have to support `XDomainRequest` for this loader. Although, if we would want this to replace the auto injection snippet for the languages it would need to support that also.

Reply:
This is also why the `_sendEvents` in this script checks for the existence of either -- it also doesn't support sending events via `XDomainRequest`, but that is just for the "unable to load the SDK" error event.

It would probably be easier to just fallback to the `ai.3.min.xxx.js` rather than the full version script in that case as it would be less code.